### PR TITLE
[LOKA] Chain API Contract

### PR DESCRIPTION
**Pull request by NicholasHartonoTvlk**
1. Chain Detail API
-> to get all information about the chain
2. Chain's Branches API
-> get All Branches from the chain (sort by location, if possible)

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -92,6 +92,58 @@ paths:
         400:
           description: "Invalide  query"
       x-swagger-router-controller: "Location"
+  /chain:
```
Comment (jyotip-sahoo) [inline swagger.yaml]: I think it should be /poi/chain

**Comment by jyotip-sahoo** [comment]
As discussed, can you please make a single api

File: swagger.yaml [hunk 2]
```diff
@@ -1309,3 +1361,162 @@ definitions:
...
+  ChainDetailResponse:
+    type: "object"
+    required:
+    - "context"
+    - "data"
+    properties:
+      data:
+        type: "object"
+        properties:
+          chainId:
```
Comment (jyotip-sahoo) [inline swagger.yaml]: we dont need chainId here, as client has information of chainId

File: swagger.yaml [hunk 3]
```diff
@@ -1309,3 +1361,162 @@ definitions:
...
+            type: "string"
+            example: "123" 
+          chainRestaurantName:
+            type: "string"
+            example: "KFC"
+          userLocation:
```
Comment (jyotip-sahoo) [inline swagger.yaml]: Do we really need UserLocation?

File: swagger.yaml [hunk 4]
```diff
@@ -1309,3 +1361,162 @@ definitions:
...
+            example: "KFC"
+          userLocation: 
+            $ref: "#/definitions/GeoDisplay"
+          totalOutletOnLocation:
+            type: "integer"
+            format: "int64"
+          mainType:
+            type: "string"
+          mainCuisine:
+            type: "string"
```
Comment (jyotip-sahoo) [inline swagger.yaml]: Please rename it to chainType and chainCuisine.
Also I need to check if we can get this two different string together or separately.

File: swagger.yaml [hunk 5]
```diff
@@ -1309,3 +1361,162 @@ definitions:
...
+      branchName:
+        type: "string"
+        example: "Slipi Kemanggisan"
+      distance:
+        type: "number"
+        example: 0.7
+      address:
+        type: "string"
+        example: "SPBU Shell, Jl. Raya S. Parman Kav 38-39, Slipi, Jakarta"
+      branchStatus:
```
Comment (jyotip-sahoo) [inline swagger.yaml]: We need to discuss on this, the current format is "<strong><font color='#00875A'>Buka</font></strong><font color='#b4b4b4'> &#8226 10:00 - 22:00 (Jumat)</font>"

File: swagger.yaml (hunk 1 above)
Comment (NicholasHartonoTvlk) [reply swagger.yaml]: hmm it should contain poi @jyotip-sahoo? I think it's different from chain feature

File: swagger.yaml (hunk 3 above)
Comment (NicholasHartonoTvlk) [reply swagger.yaml]: yes because we have to show chain based on user location

File: swagger.yaml (hunk 4 above)
Comment (NicholasHartonoTvlk) [reply swagger.yaml]: Yes, for now I will put the cuisine and type as one string

File: swagger.yaml (hunk 5 above)
Comment (NicholasHartonoTvlk) [reply swagger.yaml]: @josuaclement

Comment (NicholasHartonoTvlk) [reply swagger.yaml]: based on discussion, we will make this separated data

File: swagger.yaml (hunk 4 above)
Comment (NicholasHartonoTvlk) [reply swagger.yaml]: Based on discussion this should be 2 different field

File: swagger.yaml (hunk 2 above)
Comment (NicholasHartonoTvlk) [reply swagger.yaml]: will remove

File: swagger.yaml (hunk 3 above)
Comment (robintiolie) [reply swagger.yaml]: can rename to geoDisplay to make it consistent @NicholasHartonoTvlk

**Comment by NicholasHartonoTvlk** [comment]
> As discussed, can you please make a single api

as discussed, we will make this 2 api, 1 for chain detail, 1 for branches of the chain

File: swagger.yaml [hunk 6]
```diff
@@ -1456,6 +1506,162 @@ definitions:
...
+    - "context"
+    - "data"
+    properties:
+      data:
+        type: "object"
+        properties:
+          chainRestaurantName:
+            type: "string"
+            example: "KFC"
+          geoDisplay:
```
Comment (jyotip-sahoo) [inline swagger.yaml]: why you need geoDsplay in response? I think client has this information

**Comment by jyotip-sahoo** [comment]
LGTM

File: swagger.yaml [hunk 7]
```diff
@@ -1456,6 +1506,162 @@ definitions:
...
+        properties:
+          chainRestaurantName:
+            type: "string"
+            example: "KFC"
+          geoDisplay: 
+            $ref: "#/definitions/GeoDisplay"
+          totalOutletOnLocation:
+            type: "integer"
+            format: "int64"
+          chainType:
```
Comment (jyotip-sahoo) [inline swagger.yaml]: What will be chainType? I dont see current system supports chainType

Comment (NicholasHartonoTvlk) [reply swagger.yaml]: same as restaurant type

**Pull request merged**